util/compression_test.cc

```diff
@@ -1768,122 +1768,271 @@ TEST_F(DBAutoSkip, AutoSkipCompressionManager) {
+class AutoTuneFlushBlockPolicyFactory : public FlushBlockPolicyFactory {
  public:
-  explicit CostAwareTestFlushBlockPolicyFactory(const int window)
-      : window_(window) {}
+  explicit AutoTuneFlushBlockPolicyFactory(const int wait_block_count)
+      : wait_block_count_(wait_block_count) {}
 
   virtual const char* Name() const override {
-    return "CostAwareTestFlushBlockPolicyFactory";
+    return "AutoTuneFlushBlockPolicyFactory";
   }
 
   virtual FlushBlockPolicy* NewFlushBlockPolicy(
       const BlockBasedTableOptions& /*table_options*/,
       const BlockBuilder& data_block_builder) const override {
     (void)data_block_builder;
-    return new CostAwareTestFlushBlockPolicy(window_, data_block_builder);
+    return new AutoTuneFlushBlockPolicy(wait_block_count_, data_block_builder);
   }
 
  private:
-  int window_;
+  int wait_block_count_;
 };
-class DBCompressionCostPredictor : public DBTestBase {
+class DBAutoTuneCompressionTest : public DBTestBase {
  public:
-  Options options;
-  DBCompressionCostPredictor()
-      : DBTestBase("db_cpuio_skip", /*env_do_fsync=*/true),
-        options(CurrentOptions()) {
-    options.compression_manager = CreateCostAwareCompressionManager();
+  Options options_;
+  Random rnd;
+  int next_key_;
+  DBAutoTuneCompressionTest()
+      : DBTestBase("db_autotune", /*env_do_fsync=*/true),
+        options_(CurrentOptions()),
+        rnd(231),
+        next_key_(0) {
+    double cpu_upper_bound = 0.7;
+    double cpu_lower_bound = 0.5;
+    double io_upper_bound = 0.7;
+    double io_lower_bound = 0.5;
     auto statistics = ROCKSDB_NAMESPACE::CreateDBStatistics();
-    options.statistics = statistics;
-    options.statistics->set_stats_level(StatsLevel::kExceptTimeForMutex);
+    options_.statistics = statistics;
+    options_.statistics->set_stats_level(StatsLevel::kExceptTimeForMutex);
     BlockBasedTableOptions bbto;
     bbto.enable_index_compression = false;
     bbto.flush_block_policy_factory.reset(
-        new CostAwareTestFlushBlockPolicyFactory(10));
-    options.table_factory.reset(NewBlockBasedTableFactory(bbto));
-    DestroyAndReopen(options);
-  }
-};
-TEST_F(DBCompressionCostPredictor, CostAwareCompressorManager) {
-  // making sure that the compression is supported
-  if (!ZSTD_Supported()) {
-    return;
+        new AutoTuneFlushBlockPolicyFactory(2000));
+    options_.table_factory.reset(NewBlockBasedTableFactory(bbto));
+    options_.write_buffer_size = 19000000000;
+    options_.rate_limiter.reset(NewGenericRateLimiter(
+        1000000000, 1000 /* refill_period_us */, 10 /* fairness */,
+        RateLimiter::Mode::kWritesOnly));
+    std::shared_ptr<IOGoal> io_goal =
+        std::make_shared<IOGoal>(io_upper_bound, io_lower_bound);
+    std::shared_ptr<IOGoal> cpu_budget =
+        std::make_shared<IOGoal>(cpu_upper_bound, cpu_lower_bound);
+    options_.compression_manager = CreateAutoTuneCompressionManager(
```

Comment:
I don't think the DB is reopened with this option and the autotune compression manager is not used in the test.